### PR TITLE
Fix gradient accumulation

### PR DESCRIPTION
**Pull request by shkarupa-alex**
Fix https://github.com/keras-team/keras/issues/19990
Removes unnecessary (wrong) weight decay and gradient clipping during gradient accumulation. The will be applied only before actual variable update.

To make gradient accumulation work in the same way as "large batch" real number of iterations was hidden under property.
This is done because a lot of optimizers use it during gradient estimation in update_step.

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. It seems the tests are failing: https://github.com/keras-team/keras/actions/runs/9955833252/job/27521695775?pr=19998

Can you explain the issue with the current implementation?

**Comment by shkarupa-alex** [comment]
> Thanks for the PR. It seems the tests are failing: https://github.com/keras-team/keras/actions/runs/9955833252/job/27521695775?pr=19998

Test fixed.

> Can you explain the issue with the current implementation?

See https://github.com/keras-team/keras/issues/19990 for details.

**Review by fchollet — APPROVED** [review]
Reviewed the code -- I understand the motivation now, and I think it's a reasonable change. We can include it. Thanks for the PR.
